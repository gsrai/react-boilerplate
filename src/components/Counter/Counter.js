import React from 'react'

class Counter extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      value: 0
    }
  }
  handleIncrement = (e) => {
    this.setState({
      value: this.state.value + 1
    })
  }

  handleDecrement = (e) => {
    this.setState({
      value: this.state.value - 1
    })
  }

  render() {
    return (
      <div>
        <span>
          <button className={'counter-decrement-btn'} onClick={(e) => this.handleDecrement(e)}> - </button>
        </span>
        <span>
          {this.state.value}
        </span>
        <span>
          <button className={'counter-increment-btn'} onClick={(e) => this.handleIncrement(e)}> + </button>
        </span>
      </div>
    )
  }
}

export default Counter
