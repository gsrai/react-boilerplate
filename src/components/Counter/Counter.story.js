import React from 'react'
import { storiesOf } from '@storybook/react'
import Counter from './Counter'

storiesOf('Components / Counter', module)
  .add('default', () => <Counter />)
