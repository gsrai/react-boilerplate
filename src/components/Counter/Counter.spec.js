import { shallow } from 'enzyme'
import React from 'react'
import Counter from './Counter'

describe('Component Rendering', () => {
  test('component should render correctly', () => {
    const wrapper = shallow(<Counter />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Component Actions', () => {
  test('component should increment counter when + button is pressed', () => {
    const wrapper = shallow(<Counter />)

    expect(wrapper.state().value).toEqual(0)
    wrapper.find('.counter-increment-btn').simulate('click')
    expect(wrapper.state().value).toBe(1)
  })

  test('component should increment counter when - button is pressed', () => {
    const wrapper = shallow(<Counter />)

    expect(wrapper.state().value).toEqual(0)
    wrapper.find('.counter-decrement-btn').simulate('click')
    expect(wrapper.state().value).toBe(-1)
  })
})
