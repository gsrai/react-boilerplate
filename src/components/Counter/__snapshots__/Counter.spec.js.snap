// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Rendering component should render correctly 1`] = `
<div>
  <span>
    <button
      className="counter-decrement-btn"
      onClick={[Function]}
    >
       - 
    </button>
  </span>
  <span>
    0
  </span>
  <span>
    <button
      className="counter-increment-btn"
      onClick={[Function]}
    >
       + 
    </button>
  </span>
</div>
`;
